src/mscorlib/src/System/Text/ASCIIEncoding.cs

```diff
@@ -281,7 +302,7 @@ internal override unsafe int GetByteCount(char* chars, int charCount, EncoderNLS
             return byteCount;
         }
 
-        internal override unsafe int GetBytes(char* chars, int charCount,
+        internal override unsafe int GetBytesFallback(char* chars, int charCount,
```

Comment:
I believe this change will cause the code compiled as with BIGENDIAN to have stackoverflow (recursive calls). 

 public override int GetBytes(String chars, int charIndex, int charCount,byte[] bytes, int byteIndex) calls 
EncodingForwarder.GetBytes(this, chars, charIndex, charCount, bytes, byteIndex); which will call 
encoding.GetBytes(pChars + charIndex, charCount, pBytes + byteIndex, byteCount, encoder: null);

because you have renamed it here, the base encoding (Encoding class) version will get called which is 
Encoding.GetBytes(char* chars, int charCount, byte* bytes, int byteCount, EncoderNLS encoder) which calls 
ASCIIEncoding.GetBytes(char* chars, int charCount, byte* bytes, int byteCount) which calls 
EncodingForwarder.GetBytes(this, chars, charIndex, charCount, bytes, byteIndex); which will call 

and we'll loop forever